src/common/eos-dac/data/eos-dac.service-impl.ts

```diff
@@ -0,0 +1,112 @@
+import { Failure } from '@core/architecture/domain/failure';
+import { Result } from '@core/architecture/domain/result';
+import fetch, { RequestInit } from 'node-fetch';
+import { config } from '../../../config';
+import { RequestError } from '../../../core/architecture/data/errors/request.error';
+import { CurrencyStats } from '../domain/entities/currecy-stats';
+import { EosDacInfo } from '../domain/entities/eos-dac-info';
+import { EosDacInfoDto, CurrencyStatsDto } from './eos-dac.dtos';
+
+/**
+ * Represents the service responsible for communication with eosDAC
+ *
+ * @class
+ */
+export class EosDacServiceImpl {
```

Comment:
we have abstract classes (as interfaces) with the proper name in this case EosDacService and we map it manually to the instance of its "implementation"

Reply:
so yes we need it

Reply:
Because we use inversify and like I wrote we map abstracts to implementation, I dont want to use IFoo as interface... Ive seen earlier people dont like that, and here we nicely inject impl by name like FooRepository and not IFoorepository